File: java_code/warmup2/stringMatch.java
package java_code.warmup2;

public class stringMatch {

    public int stringMatch(String a, String b) {

        int length;
        if (a.length() < b.length())
            length = a.length();
        else
            length = b.length();

        int count = 0;

        for (int i = 1; i < length; i++) {
            if (a.charAt(i - 1) == b.charAt(i - 1) && a.charAt(i) == b.charAt(i)) {
                count++;
            }
        }

        return count;
    }

}

File: python/warmup2/string_match.py
#codingbat link: https://codingbat.com/prob/p182414


def string_match(a, b):
    if (len(a) == 0 or len(b) == 0):
        return 0

    count = 0
    if (len(a) < len(b)):
        for i in range(len(a) - 1):
            if (a[i:i + 2] == b[i:i + 2]):
                count += 1

    else:
        for i in range(len(b) - 1):
            if (a[i:i + 2] == b[i:i + 2]):
                count += 1

    return count